components/lib/crash/src/test/java/mozilla/components/lib/crash/service/MozillaSocorroServiceTest.kt

```diff
@@ -24,27 +25,71 @@ class MozillaSocorroServiceTest {
             testContext,
             "Test App"
         ))
-        doNothing().`when`(service).sendViaGeckoViewCrashReporter(any())
+        doNothing().`when`(service).sendReport(any(), any(), any())
 
         val crash = Crash.NativeCodeCrash("", true, "", false, arrayListOf())
         service.report(crash)
 
         verify(service).report(crash)
-        verify(service).sendViaGeckoViewCrashReporter(crash)
+        verify(service).sendReport(null, crash.minidumpPath, crash.extrasPath)
     }
 
     @Test
-    fun `MozillaSocorroService does not send uncaught exception crashes`() {
+    fun `MozillaSocorroService send uncaught exception crashes`() {
         val service = spy(MozillaSocorroService(
             testContext,
             "Test App"
         ))
-        doNothing().`when`(service).sendViaGeckoViewCrashReporter(any())
+        doNothing().`when`(service).sendReport(any(), any(), any())
 
         val crash = Crash.UncaughtExceptionCrash(RuntimeException("Test"), arrayListOf())
         service.report(crash)
 
         verify(service).report(crash)
-        verifyNoMoreInteractions(service)
+        verify(service).sendReport(crash.throwable, null, null)
+    }
+
+    @Test
+    fun `MozillaSocorroService parses extrasFile correctly`() {
+        val service = spy(MozillaSocorroService(
+                testContext,
+                "Test App"
+        ))
+        val file = File(getResource("TestExtrasFile").file)
+        val extrasMap = service.readExtrasFromFile(file)
+
+        assert(extrasMap.size == 9)
+        assert(extrasMap["InstallTime"] == "1569440259")
+        assert(extrasMap["ProductName"] == "Test Product")
+        assert(extrasMap["StartupCrash"] == "0")
+        assert(extrasMap["StartupTime"] == "1569642043")
+        assert(extrasMap["useragent_locale"] == "en-US")
+        assert(extrasMap["CrashTime"] == "1569642098")
+        assert(extrasMap["UptimeTS"] == "56.3663876")
+        assert(extrasMap["SecondsSinceLastCrash"] == "104")
+        assert(extrasMap["TextureUsage"] == "12345678")
+    }
+
+    @Test
+    fun `MozillaSocorroService unescape strings correctly`() {
+        val service = spy(MozillaSocorroService(
+                testContext,
+                "Test App"
+        ))
+        val test1 = "\\\\\\\\"
+        val expected1 = "\\"
+        assert(service.unescape(test1) == expected1)
+
+        val test2 = "\\\\n"
+        val expected2 = "\n"
+        assert(service.unescape(test2) == expected2)
+
+        val test3 = "\\\\t"
+        val expected3 = "\t"
+        assert(service.unescape(test3) == expected3)
+
+        val test4 = "\\\\\\\\\\\\t\\\\t\\\\n\\\\\\\\"
+        val expected4 = "\\\t\t\n\\"
+        assert(service.unescape(test4) == expected4)
     }
```

Comment:
I would love to see a test here that spins up a `MockWebServer` and tests/verifies the whole thing (maybe even some error cases?).

Reply:
OK. I'll look into this.

Reply:
Please let me know the two new tests are what you were looking for.

Reply:
You can get the request the mockWebServer retrieved to analyze it too. Maybe we can do a quick sanity check that it contains some of the things we expect it to?

Reply:
Sure, I'll try that.  Thanks.

Reply:
Got request unzipped into text.  Just using String.contains to find some of the consistent values that we expect to see.  Let me know what you think.  Thanks,